naga/src/front/wgsl/lower/mod.rs

```diff
@@ -2541,12 +2542,76 @@ impl<'source, 'temp> Lowerer<'source, 'temp> {
                         "select" => {
                             let mut args = ctx.prepare_args(arguments, 3, span);
 
-                            let reject = self.expression(args.next()?, ctx)?;
-                            let accept = self.expression(args.next()?, ctx)?;
+                            let reject_orig = args.next()?;
+                            let accept_orig = args.next()?;
+                            let mut values = [
+                                self.expression_for_abstract(reject_orig, ctx)?,
+                                self.expression_for_abstract(accept_orig, ctx)?,
+                            ];
                             let condition = self.expression(args.next()?, ctx)?;
 
                             args.finish()?;
 
+                            fn expr_ty_res<'a>(
+                                ctx: &'a ExpressionContext<'_, '_, '_>,
+                                expr: Handle<ir::Expression>,
+                            ) -> &'a proc::TypeResolution {
+                                &ctx.typifier()[expr]
+                            }
```

Comment:
This function doesn't earn its keep. See notes below.

Reply:
I agree; in previous versions of this PR, this was a more complicated body, but now it's trivial to as to not be worth it.

Fixed as suggested with 68a56cf4. LSP made this _trivial_ to inline. 😀